### ISSUE TITLE
How do I edit a file inside the egg file? (translating urls names)

### ISSUE THREAD

**Issue by h-2-0** [macropin/django-registration#43]
Hello!

I'm in the process of translating url names, templates, etc..and I'd like to know if it would be a good idea to modify the urls.py and auth_urls.py  files residing inside the egg (I've installed django-registration-redux using 'sudo pip install...')?

/usr/local/lib/python3.2/dist-packages/django_registration_redux-1.2-py3.2.egg 

Or what about unzipping the egg, modify files, then rezippit again? If so, how that works? Or what about just unziping it? If so, how do I name the folder inside ../dist-packages/?   

 This problem has stumped me over the last two days. I have little idea on what to do. I'm so confused. In any case, am I going in the right direction?

This is my root urls.py
    url(r'^cuenta/', include('django.contrib.auth.urls')), 
    url(r'^cuenta/', include('registration.backends.default.urls')),

Thanks in advance!

**Comment by chhantyal** [comment]
If you just want to override urls, then just do it in your project urls.py file and put that before django-registration's urls.

**Comment by h-2-0** [comment]
Thanks, chhantyal!

I added extra urlpatterns within my root urls.py. One stemming from auth_urls.py and the other on from django.backends.default.urls. Now my root url.py is a huge mess but I guess that's better than nothing!

In any case, why on earth, django-registration-redux is installed as an egg file? Why not as a folder? !It's so annoying for a newbie like me! It took me hours to trial and error to realise the files I wanted to edit weren't inside any of my Python folders (I had used pip2.7, pip3.2 and pip to install it everywhere on my system). Documentation doesn't say anything about it. No one in the entire web says anything about it. And when I finally realised the package was inside a Python Egg, I had to stop the whole development process to research and learn about them in order to modify them. Well, they say one learns something new every day, but, unfortunately that's what I learned that day.

But rant  is over! Thanks again everyone!

tl;dr Egg folder, not file please!

**Comment by chhantyal** [comment]
Why is it mess?
It should look like
````
url('^overide/', 'view_name', )
url(r'^cuenta/', include('django.contrib.auth.urls')), 
url(r'^cuenta/', include('registration.backends.default.urls')),
````

Any python package is installed like that. It's simply way of distribution python library, just like RPM or deb. Anyway, that's not at all related to this package and you can read on internet to understand about them. 

Please close this issue and only discuss things that are related to this package.

**Comment by h-2-0** [comment]
Hi, chhantyal!

Thanks again for replying. 

The reason I started this thread is because I wanted to translate url paths from this file:
    `r'^cuenta/', include('registration.backends.default.urls')),`
...which resides in an egg zip file.  

Since I couldn't I ended solving adding this to my root urls.py:
`registration_urlpatterns = patterns('',url(r'^activacion/completa/$',TemplateView.as_view(template_name='registrationactivation_complete.html'),name='registration_activation_complete'),`
`...`
`...`
`url(r'^cuenta/', include(registration_urlpatterns)),`
`if getattr(settings, 'INCLUDE_REGISTER_URL', True):`
etc..

I shall now close this thread. Many thanks !

**Comment by derek-adair** [comment]
FYI its never a good idea to... EVER modify a file in an egg.  It defies the point of a distribution.  you need to look into [pypi.org](https://pypi.python.org/pypi) and how to package your code.  Also the django tutorial is very thurough at explaining [reusable code](https://docs.djangoproject.com/en/1.8/intro/reusable-apps/) (which is a concept that you should find useful)

**Comment by h-2-0** [comment]
Thanks, derek-adar! I should put my own eggs next time to keep things open source. Making your own web app package sounds like a great idea too! I've read the 'reusable code' headline before in the Django tutorial but never read the article until today. Thanks everybody!
